test/sample_delcache.sh

```diff
@@ -34,60 +34,59 @@ if [ "X$1" = "X-h" -o "X$1" = "X-H" ]; then
 fi
 if [ "X$1" = "X" -o "X$2" = "X" -o "X$3" = "X" ]; then
   func_usage $PRGNAME
-  exit -1
+  exit 1
 fi
 
 BUCKET=$1
-CDIR=$2
+CDIR="$2"
 LIMIT=$3
 SILENT=0
 if [ "X$4" = "X-silent" ]; then
   SILENT=1
 fi
-FILES_CDIR=$CDIR/$BUCKET
-STATS_CDIR=$CDIR/\.$BUCKET\.stat
-
+FILES_CDIR="${CDIR}/${BUCKET}"
+STATS_CDIR="${CDIR}/.${BUCKET}.stat"
+CURRENT_CACHE_SIZE=`du -sb "$FILES_CDIR" | awk '{print $1}'`
 #
 # Check total size
 #
-if [ $LIMIT -ge `du -sb $FILES_CDIR | awk '{print $1}'` ]; then
+if [ $LIMIT -ge $CURRENT_CACHE_SIZE ]; then
   if [ $SILENT -ne 1 ]; then
-    echo "$FILES_CDIR is below allowed $LIMIT"
+    echo "$FILES_CDIR ($CURRENT_CACHE_SIZE) is below allowed $LIMIT"
   fi
   exit 0
 fi
 
-#
-# Make file list by sorted access time
-#
-ALL_STATS_ATIMELIST=`find $STATS_CDIR -type f -exec echo -n {} \; -exec echo -n " " \; -exec stat -c %X {} \; | awk '{print $2":"$1}' | sort`
-
 #
 # Remove loop
 #
 TMP_ATIME=0
 TMP_STATS=""
 TMP_CFILE=""
-for part in $ALL_STATS_ATIMELIST; do
-  TMP_ATIME=`echo $part | sed 's/\:/ /' | awk '{print $1}'`
-  TMP_STATS=`echo $part | sed 's/\:/ /' | awk '{print $2}'`
-  TMP_CFILE=`echo $TMP_STATS | sed s/\.$BUCKET\.stat/$BUCKET/`
-
-  if [ `stat -c %X $TMP_STATS` -eq $TMP_ATIME ]; then
-    rm -f $TMP_STATS $TMP_CFILE > /dev/null 2>&1
+#
+# Make file list by sorted access time
+#
+find "$STATS_CDIR" -type f -exec stat -c "%X:%n" "{}" \; | sort | while read part
```

Comment:
Could you simplify this logic with:

```
find -printf '%A@\t%p\n' | sort -n
```

Reply:
Kinda. You could use `-printf '%A@:%p\n'` (you need the `:` as a separator for the `cut` command that comes later), but this `printf` gives you fractional seconds:

```
1554192949.6842425020:/mnt/.s3-cache/.my-bucket.stat/some/file with a space
```

but the existing comparison code expects to be fed plain-old integer epoch seconds. If you could get the `printf` to drop the fractions, you'd be good.